clang/include/clang/Lex/HeaderSearch.h

```diff
@@ -84,7 +84,9 @@ struct HeaderFileInfo {
   LLVM_PREFERRED_TYPE(bool)
   unsigned isModuleHeader : 1;
 
-  /// Whether this header is a `textual header` in a module.
+  /// Whether this header is a `textual header` in a module. If a header is
+  /// textual in one module and normal in another module, this bit will not be
+  /// set, only `isModuleHeader`.
```

Comment:
Maybe this behavior is weird? Maybe both bits should be set in this scenario? `HeaderSearch::ShouldEnterIncludeFile` -> `MaybeReenterImportedFile` could change its check to `!FileInfo. isModuleHeader && FileInfo.isTextualModuleHeader`

Reply:
I don't have a strong preference here.